src/main/mi.mc

```diff
@@ -6,37 +6,66 @@
 include "compile.mc"
 include "seq.mc"
 include "string.mc"
+include "eval.mc"
 include "run.mc"
 include "assoc.mc"
 include "options.mc"
 
 mexpr
 
 -- Menu
-let menu = strJoin "\n" [
-  "Usage: mi [compile|run] <files>",
-  "",
-  "Options:",
-  "  --debug-parse                    Print the AST after parsing",
-  "  --debug-generate                 Print the AST after code generation",
-  "  --exit-before                    Exit before evaluation or compilation",
-  "  --test                           Generate utest code",
-  ""]
+let menu =
+"Usage: mi <command> file [<options>|<args>]
+
+Commands:
+  eval      Evaluates a .mc file using an internal interpreter
+  compile   Compiles a .mc file into an executable with the same name
+  run       Combines eval and compile, to run the program as fast as possible
+  test      Same as run, but with tests enabled and where the texts after
+            the .mc file are arguments to the executable program
+
+If no command is given, the file will be executed using the run command
+and all arguments after the file are arguments to the .mc executed file.
+
+Options:
+  --debug-parse       Print the AST after parsing
+  --debug-generate    Print the AST after code generation
+  --exit-before       Exit before evaluation or compilation
+  --test              Generate utest code
+  -- [args]           If the run or eval commands are used, then the texts
+                      following -- are arguments to the executed program
+"
```

Comment:
How come there is both a `test` command and a `--test` flag now? What is the difference between them?

Reply:
The idea was to use the test command where you can give arguments to the actual program (not to `mi`) directly. E.g.

```
mi test hi.mc --foo
```
means that `--foo` is a parameter to `hi.mc` and not to `mi`. 

An alternative is to shadow certain parameters from `mi`, so that the they take over. In such a case the `test` command is not needed. Let's discuss the alternatives at the Miking meeting.